src/timepicker/TimePicker.ts

```diff
@@ -0,0 +1,359 @@
+ * @property onBlur             Called when the input is blurred
+ * @property onChange           Called when the value changes
+ * @property onFocus            Called when the input is focused
+ * @property onMenuChange       Called when menu visibility changes
+ * @property onRequestOptions   Called when options are shown; should be used to set `options`
+ * @property openOnFocus        Determines whether the result list should open when the input is focused
+ * @property options            Options for the current input; should be set in response to `onRequestOptions`
+ * @property readOnly           Prevents user interaction
+ * @property required           Determines if this input is required, styles accordingly
+ * @property start              The minimum time to display in the menu (defaults to '00:00:00')
+ * @property step               The number of seconds between each option in the menu (defaults to 60)
+ * @property useNativeElement   Use the native <input type="time"> element if true
+ * @property value              Value to set on the input
+ */
+export interface TimePickerProperties extends ThemeableProperties {
+	autoBlur?: boolean;
+	clearable?: boolean;
+	customOptionItem?: any;
+	customOptionMenu?: any;
+	disabled?: boolean;
+	end?: string;
+	formId?: string;
+	getOptionLabel?(option: TimeUnits): string;
+	inputProperties?: TextInputProperties;
+	invalid?: boolean;
+	isOptionDisabled?(result: any): boolean;
+	label?: string | LabelOptions;
+	name?: string;
+	onBlur?(value: string): void;
+	onChange?(value: string): void;
+	onFocus?(value: string): void;
+	onMenuChange?(open: boolean): void;
+	onRequestOptions?(value: string, getOptions: () => TimeUnits[]): void;
+	openOnFocus?: boolean;
+	options?: TimeUnits[];
+	readOnly?: boolean;
+	required?: boolean;
+	start?: string;
+	step?: number;
+	useNativeElement?: boolean;
+	value?: string;
+}
+
+export interface TimeUnits {
+	hour: number;
+	minute?: number;
+	second?: number;
+}
+
+/**
+ * @private
+ * Regular epression that matches a standard time string ('HH:mm:ss').
+ */
+const TIME_PATTERN = /^\d{2}:\d{2}(:\d{2})?$/;
+
+/**
+ * Generate an array of time unit objects from the specified start date to the specified end date.
+ *
+ * @param start    The start time. Defaults to midnight.
+ * @param end      The end time. Defaults to 23:59:59.
+ * @param step     The amount of time in seconds between each step. Defaults to 60.
+ * @return         An array of time unit objects.
+ */
+export function getOptions(start: string = '00:00:00', end: string = '23:59:59', step = 60): TimeUnits[] {
+	const endUnits = parseUnits(end);
+	const startUnits = parseUnits(start);
+	const endTime = getTime(endUnits);
+	const date = new Date();
+	let time = getTime(startUnits, date);
+
+	const i = step * 1000;
+	const options: number[] = [];
+
+	while (time <= endTime) {
+		options.push(time);
+		time += i;
+	}
+
+	return options.map((time: number) => {
+		const date = new Date(time);
```

Comment:
I might be able to reuse the same `Date` object so I'm not instantiating one each time.